[peiran18] iP

peiran18 opened the pull request:
### PotatoPro

> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

PotatoPro frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- _SUPER FAST_ to use

All you need to do is,

1. download it from [here](https://nus-cs2103-ay2324s1.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE**!

Features:

- [ ] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

If you Java programmer, you can use it to practice Java too. Here's the `main` method:

```ruby
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

peiran18 committed:
Potato greets user and exits

peiran18 committed:
Echoes user commands until bye, added horizontal lines

peiran18 committed:
Store and display user input

peiran18 committed:
Added Task class, able to mark task as done

peiran18 committed:
track todo, deadline, event

peiran18 committed:
identify todo, deadline, event using keywords

peiran18 committed:
minimal exceptions handled

peiran18 committed:
minor fixes

peiran18 committed:
delete tasks

peiran18 committed:
testing

peiran18 committed:
all bugs fixed

peiran18 committed:
Add saving feature

peiran18 committed:
Teach the chatbot how to understand dates and times

peiran18 committed:
Merge branch 'branch-Level-8'

# Conflicts:
#	src/main/java/Deadline.java
#	src/main/java/Event.java
#	src/main/java/Task.java

peiran18 committed:
Merge all previous branches

peiran18 committed:
Add data saving and loading

peiran18 committed:
Merge branch 'branch-Level-7'

peiran18 committed:
Add date identification feature

peiran18 committed:
Merge branch 'branch-Level-8'

craigtonlian commented on src/main/java/DateTime.java:
```diff
@@ -0,0 +1,16 @@
+import java.time.temporal.ChronoUnit;
+public class DateTime {
+    private LocalDate date;
+//    private LocalTime time;
```
Maybe this can be removed?

craigtonlian commented on src/main/java/DateTime.java:
```diff
@@ -0,0 +1,16 @@
+    public DateTime(String date) {
+        this.date = LocalDate.parse(date);
+    }
+
```
I noticed your DateTime class only parses dates. Perhaps that can reflected in the class name?

craigtonlian commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,38 @@
+                    input.substring(indexBy + 4));
+        }
+    }
+
```
Why does this look so similar to the `deadlineParse` function? Perhaps an abstraction can help to reduce duplicate code.

craigtonlian commented on src/main/java/Event.java:
```diff
@@ -0,0 +1,42 @@
+                    input.substring(indexTo + 4));
+        }
+    }
+
```
I noticed something similar in the other `Task` subclasses. Maybe adding a boolean `isSaved` as a parameter can help to remove duplicate code.

craigtonlian commented on src/main/java/Potato.java:
```diff
@@ -0,0 +1,124 @@
+    public void list(ArrayList<Task> tasks) {
+        int count = 0;
+        System.out.println(LINE + "Ok look all you want but they literally won't do themselves?");
+
```
Love the sass.

craigtonlian commented on src/main/java/Todo.java:
```diff
@@ -0,0 +1,22 @@
+    public String toString() {
+        return "[T]" + super.toString();
+    }
+}
```
Love how clean this code is! 👍

craigtonlian reviewed (commented):
I like this PR! Keep it up! 👍

craigtonlian commented on src/main/java/Potato.java:
```diff
@@ -0,0 +1,124 @@
+            new PotatoException(LINE +
+                    "☹ OOPS!!! I'm sorry, but I don't know what that means :-(\n" +
+                    LINE);
+
```
Did you intend to throw this error instead of just instantiating it?

craigtonlian commented on src/main/java/Todo.java:
```diff
@@ -0,0 +1,22 @@
+    public static Todo todoParse(String input) {
+        if (input.length() < 1) {
+            new PotatoException(LINE + "Bruh you wanna do air or something?\n" + LINE);
+            return null;
```
Maybe if you threw the `PotatoException` then you can avoid returning null?

craigtonlian reviewed (commented):
Maybe `Exception`s can be thrown to avoid returning null in some places?

peiran18 committed:
Add OOP

peiran18 committed:
Merge branch 'branch-MoreOOP'

peiran18 committed:
Add gradle

peiran18 committed:
Add JUnit

peiran18 committed:
Create Jar File

peiran18 committed:
Fix bugs

peiran18 committed:
Add Javadocs

peiran18 committed:
Make changes according to coding standard

peiran18 committed:
Add find feature

peiran18 committed:
Merge branch 'branch-A-JavaDoc'

peiran18 committed:
Merge branch 'branch-A-CodingStandard'

peiran18 committed:
Merge branch 'Level-9'

peiran18 committed:
Fix minor bugs

peiran18 committed:
Merge branch 'master' of https://github.com/peiran18/ip

# Conflicts:
#	src/main/java/DateTime.java
#	src/main/java/Deadline.java
#	src/main/java/Event.java
#	src/main/java/Potato.java
#	src/main/java/Storage.java
#	src/main/java/Task.java
#	src/main/java/Todo.java

seraphimstreets commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,94 @@
+    public static Task savedParse(String input) {
+<<<<<<< HEAD
+        String[] parts = input.split(" \\| ");
+        System.out.println(parts[0]);
```
You should probably remove these in future iterations.

seraphimstreets commented on src/main/java/Event.java:
```diff
@@ -0,0 +1,61 @@
+    protected DateTime start;
+    protected DateTime end;
+
+<<<<<<< HEAD
```
I notice you have a few of these unremoved git conflict markers. You'll have to delete these manually, otherwise the code won't be functional!

seraphimstreets commented on src/main/java/Event.java:
```diff
@@ -0,0 +1,61 @@
+        }
+    }
+
+    public static Event eventSavedParse(String input) {
```
The method name could probably be rearranged to be clearer. For instance, parseSavedEvent.

seraphimstreets commented on src/main/java/Potato.java:
```diff
@@ -0,0 +1,166 @@
+    public void run() {
+        ui.showWelcome();
+        boolean isExit = false;
+        ui.showLine(); // show the divider line ("_______")
```
this comment could probably be removed since the method name is quite clear on what it does.

seraphimstreets commented on src/main/java/Potato.java:
```diff
@@ -0,0 +1,166 @@
+        System.out.println(LINE);
+    }
+
+    public void add(String input, ArrayList<Task> tasks) throws IOException {
```
Again, all these method names could be more informative eg. addTask.

seraphimstreets commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,94 @@
+        return (isDone ? "X" : " ");
+    }
+
+    public void setStatus(boolean bool) {
```
the code standard is to name setter methods after the variable eg. setIsDone

seraphimstreets reviewed (commented):
All in all, while the codebase is still currently somewhat messy, the structure is there and I liked the amusing personaliation! That said, do try to prioritize method and class name clarity over humour for later submissions.

peiran18 committed:
Make improvements based on peer review

peiran18 committed:
Add package refinements

peiran18 committed:
Merge commit 'ed6d4d25bd4e34461919ffe785a6f10f4c087552'

peiran18 committed:
Fix bugs and gradle

peiran18 committed:
Add more tests

peiran18 committed:
Add base case for find feature

peiran18 committed:
Add checkstyle files and fix code according to coding standards

peiran18 committed:
Merge branch 'branch-A-checkstyle'

peiran18 committed:
Implement functional GUI

peiran18 committed:
Fix GUI bugs

peiran18 committed:
Implement GUI using FXML

peiran18 committed:
Merge branch 'branch-Level-10'

peiran18 committed:
Find command: make matching case-insensitive

Find command is case-sensitive.

A case-insensitive find is more user-friendly because users cannot be
expected to remember the exact case of the keywords.

Let's update the search algorithm to use case-insensitive matching.

peiran18 committed:
FXML goodbye: close application

command "bye" prints goodbye UI but does not exit app.

The application should close automatically upon "bye" command.

Let's update the exitCommand to exit the application.

peiran18 committed:
Add assertions

peiran18 committed:
Merge pull request #2 from peiran18/branch-A-Assertions

Add assertions

peiran18 committed:
Merge branch 'master' of https://github.com/peiran18/ip

peiran18 committed:
Improve code quality

peiran18 committed:
Merge pull request #3 from peiran18/branch-A-CodeQuality

Improve code quality

peiran18 committed:
Merge branch 'master' of https://github.com/peiran18/ip

# Conflicts:
#	src/main/java/potato/Main.java
#	src/main/java/potato/MainWindow.java
#	src/main/java/potato/Potato.java

peiran18 committed:
Add priority feature to label tasks

peiran18 committed:
Merge branch 'branch-A-priority'

peiran18 committed:
Fix hello and goodbye commands

peiran18 committed:
Add screenshot of UI

peiran18 committed:
Add readme

peiran18 committed:
Improve documention

peiran18 committed:
Minor bug fix

peiran18 committed:
Fix exceptions

peiran18 committed:
Add more exceptions

peiran18 committed:
Add clear function

peiran18 committed:
Clean up based on style check

peiran18 committed:
Fix minor bugs
